src/robotFactory.js

```diff
@@ -1,10 +1,75 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    Object.keys(coords).length > 1
+      ? this.coords = coords
+      : this.coords = {
+        x: 0, y: 0,
+      };
```

Comment:
First of all, use default values for your function params
![image](https://user-images.githubusercontent.com/37161315/175529001-c9ae29ec-6060-4004-bf5a-3ade5249b879.png)

Reply:
Second, don't repeat yourself, with assignment, transform it into something like this
```suggestion
    this.coords = Object.keys(coords).length > 1
      ? coords
      :  {
        x: 0,
        y: 0,
      };
```